config/initializers/01_autosave_association_patch.rb

```diff
@@ -28,11 +28,11 @@ module AutosaveAssociation
     private
       alias old_association_valid? association_valid?
 
-      def association_valid?(reflection, record, index=nil)
+      def association_valid?(reflection, record)
```

Comment:
To avoid similar issues in the future with the change in arguments;

```suggestion
      def association_valid?(reflection, ...)
```

(and similar below)